server/game/cards/13.4-BtRK/TunnelsOfTheRedKeep.js

```diff
@@ -0,0 +1,23 @@
+const DrawCard = require('../../drawcard.js');
+
+class TunnelsOfTheRedKeep extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Kneel and return to shadows',
+            cost: ability.costs.kneelSelf(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                context.player.putIntoShadows(this, false);
+                this.game.addMessage('{0} kneels {1} to return it to shadows.', this.controller, this);
+                this.untilEndOfPhase(ability => ({
+                    match: card => card.getType() === 'character' && card.controller === this.controller,

```

Comment:
Lasting effects only apply to the cards that were in play at the time it began. So you'll need to create an array of these cards and then pass the array into `match` instead of a function. Look at [Nymeria Sand](https://github.com/throneteki/throneteki/blob/master/server/game/cards/02.2-TRtW/NymeriaSand.js#L15-L24) as an example (but use `context.player` instead of `this.controller`).

Reply:
Thank you for your great feedback. I committed the requested changes

Reply:
Just asking for clarity on the workflow here: Am i supposed to resolve the conversation now after i committed the changes or do you do that as the reviewer? Thanks!

Reply:
I don't pay much attention to whether the conversation is resolved, but you could click it after you've pushed up requested changes.